src/components/Drawer/PDrawer.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <teleport to="body">
+    <transition name="p-drawer__slide" :duration="350">
+      <keep-alive>
+        <PLayoutResizable v-if="modelValue" :disabled="!resizable" class="p-drawer" :class="classes.root" :placement="placement">
+          <template #aside>
+            <div class="p-drawer__aside" v-bind="attrs">
+              <slot v-bind="drawerScope" />
+            </div>
+          </template>
+
+          <slot name="overlay" v-bind="drawerScope">
+            <div class="p-drawer__overlay" @click="drawerScope.close" />
+          </slot>
+        </PLayoutResizable>
+      </keep-alive>
+    </transition>
+  </teleport>
+</template>
+
+<script lang="ts">
+  export default {
+    name: 'PDrawer',
+    expose: [],
+    inheritAttrs: false,
+  }
+</script>
+
+<script lang="ts" setup>
+  import { computed, useAttrs, ref } from 'vue'
+  import { useDrawer, useGlobalEventListener } from '@/compositions'
+  import { PLayoutResizable } from '@/layouts'
+  import { keys } from '@/types'
+  import { isKeyEvent } from '@/utilities'
+
+  export type PDrawerPlacement = 'left' | 'right' | 'top' | 'bottom'
+
+  const props = defineProps<{
+    open?: boolean,
+    resizable?: boolean,
+    placement?: PDrawerPlacement,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'update:open', value: boolean): void,
+  }>()
+
+  const placement = computed(() => props.placement ?? 'left')
+
+  const attrs = useAttrs()
+
+  const internalValue = ref<boolean>(false)
+  const modelValue = computed<boolean>({
```

Comment:
Can we name these `internalOpen` and `open`? We've been overriding the prop name with any computed get/sets like this.